src/good_fences_runner.rs

```diff
@@ -28,11 +28,12 @@ impl GoodFencesRunner {
         tsconfig_paths_json: TsconfigPathsJson,
         directory_paths_to_walk: &Vec<&str>,
         external_fences: ExternalFences,
+        ignored_dirs: &Vec<regex::Regex>,
     ) -> GoodFencesRunner {
         // find files
         let walked_files = directory_paths_to_walk
             .iter()
-            .map(|path| discover_fences_and_files(path, external_fences))
+            .map(|path| discover_fences_and_files(path, external_fences, ignored_dirs.clone()))
```

Comment:
I think technically we could avoid this clone with a lifetime annotation on the callback function, but that sounds complicated. This is fine as is.